allennlp/data/fields/label_field.py

```diff
@@ -35,18 +36,26 @@ class LabelField(Field[numpy.ndarray]):
         If your labels are 0-indexed integers, you can pass in this flag, and we'll skip the indexing
         step.  If this is ``False`` and your labels are not strings, this throws a ``ConfigurationError``.
     """
+    # It is possible that users want to use this field with a namespace which uses OOV/PAD tokens.
+    # This warning will be repeated for every instantiation of this class (i.e for every data
+    # instance), spewing a lot of warnings so this class variable is used to only log a single
+    # warning per namespace.
+    _should_warn_for_namespace: DefaultDict[str, bool] = defaultdict(lambda: True)
+
     def __init__(self,
                  label: Union[str, int],
                  label_namespace: str = 'labels',
                  skip_indexing: bool = False) -> None:
         self.label = label
         self._label_namespace = label_namespace
         self._label_id = None
-        if not (self._label_namespace.endswith("labels") or self._label_namespace.endswith("tags")):
+        if not (self._label_namespace.endswith("labels") or self._label_namespace.endswith(
```

Comment:
This is a pretty messy line.  I'd recommend simplifying this to:

```python
def __init__(self, ...):
  if self._should_warn_for_namespace(label_namespace):
    ...

def _should_warn_for_namespace(self, label_namespace):
  if label_namespace.endswith('labels') or label_namespace.endswith('tags'):
    if label_namespace not in self._already_warned_namespaces:
      logger.warning("...")
      self._already_warned_namespaces.add(label_namespace)
```

This has the benefit of also moving the multi-line warning statement out of the constructor, which will make this class easier to read.